openid_connect.server_conf

```diff
@@ -66,6 +69,70 @@
         error_page 500 502 504 @oidc_error;
     }
 
+    #
+    # User information endpoint for the following purposes:
+    # - Browser to periodically check if your are signed-in based on status code.
+    # - Browser to show the signed-in user information.
+    # - https://openid.net/specs/openid-connect-core-1_0.html#UserInfo
+    #
+    location = /userinfo {
```

Comment:
We cannot say that we support the Userinfo endpoint until such time as:
1. These requests must be made through the back channel communication (client/RP/nginx <-> Userinfo).
2. We need to parse JSON response.
3. Perform response validation in accordance with the requirement.
4. Save keys and values in a keyval.
At the same time, we are not required to validate the Access Token to perform this request.

Reply:
- Yeah, I originally thought and tried your approach.
- But, the requirement of NMS-ACM was to proxy and return the IdP's response itself to the SPA back then.
- Yeah, it is good point of access token validation too. If we validate it using local JWK file in the backend, and if it isn't invalid, then it reduces latency. However, if it is valid, then the IdP validates it using Bearer token again. 
- I will try achieving both requirements between NGINX Plus and NMS-ACM. Afterwards, you can review, and I will revise more things.  Thanks!

Reply:
- For the use case where SPA wants to show user profile in detail like `www.example.com "sub,name,preferred_username,given_name,family_name,email,photo"`, I have extracted only configured claims in `$oidc_userinfo_response_data` and return the JSON rather than directly returning all claims from the OP.
- Let me know if you have other thoughts.